include/program_options_utils.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#ifndef DISKANN_PROGRAM_OPTIONS_UTILS_CPP
+#define DISKANN_PROGRAM_OPTIONS_UTILS_CPP
+
+#include <string.h>
+
+namespace program_options_utils
+{
+const std::string make_program_description(const char *executable_name, const char *description)
+{
+    return std::string("\n")
+        .append(description)
+        .append("\n\n")
+        .append("Usage: ")
+        .append(executable_name)
+        .append(" [OPTIONS]");
+}
+
+// Required parameters
+const char *DATA_TYPE_DESCRIPTION = "data type, one of {int8, uint8, float} - float is single precision (32 bit)";
+const char *DISTANCE_FUNCTION_DESCRIPTION =
+    "distance function <l2/mips/fast_l2/cosine>.  'fast l2' and 'mips' only support data_type float";
+const char *INDEX_PATH_PREFIX_DESCRIPTION = "Path prefix to the index, e.ge '/mnt/data/my_ann_index'";
+const char *RESULT_PATH_DESCRIPTION =
+    "Path prefix for saving results of the queries, e.g. '/mnt/data/query_file_X.bin'";
+const char *QUERY_FILE_DESCRIPTION = "Query file in binary format, e.g. '/mnt/data/query_file_X.bin'";
+const char *NUMBER_OF_RESULTS_DESCRIPTION = "Number of neighbors to be returned";
+const char *SEARCH_LIST_DESCRIPTION =
+    "List of L values of search.  L value is the number of neighbor/distance pairs to keep in-memory at the same time "
+    "during a query.  The higher the value the more accurate the results and the slower the search.  This value MUST "
+    "be at least K (number of neighbors to return).";
+const char *INPUT_DATA_PATH = "Input data file in bin format.  This is the file you want to build the index over.";
+
+// Optional parameters
+const char *FILTER_LABEL_DESCRIPTION =
+    "Filter to use when running a query.  'filter_label' and 'query_filters_file' are mutually exclusive.";
+const char *FILTERS_FILE_DESCRIPTION =
+    "Filter file for Queries for Filtered Search.  File format is text with one filter per line.  File must "
+    "have exactly one filter OR the same number of filters as there are queries in the 'query_file'.";
+const char *LABEL_TYPE_DESCRIPTION =
+    "Storage type of Labels <uint/ushort>, default value is uint which will consume memory 4 bytes per filter";
+const char *GROUND_TRUTH_FILE_DESCRIPTION =
+    "ground truth file for the queryset"; // what's the format, what's the requirements? does it need to include an
+                                          // entry for every item or just a small subset? I have so many questions about
+                                          // this file
+const char *NUMBER_THREADS_DESCRIPTION = "Number of threads used for building index.  Defaults to number of logical "
+                                         "processor cores on your this machine returned by omp_get_num_procs()";
+const char *FAIl_IF_RECALL_BELOW = "If set to a value >0 and <100%, program returns -1 if best recall found is below "
```

Comment:
FAIL not FAIl

Reply:
Also, maybe the description should be "Value between 0 and 100, indicating the recall tolerance percentage threshold before program fails with a non-zero exit code. "

Also, if we return the results even if it results in a non-zero exit code, we should note that you'll get the results written out as well, and the exit code is intended to be treated as a signal that it did not meet your recall requirements for use (presuming that is what we do, that is).

Reply:
I verified that is what we do :).